bundles/org.eclipse.cdt.lsp.clangd/OSGI-INF/org.eclipse.cdt.lsp.internal.clangd.ClangdMetadataDefaults.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.3.0" name="org.eclipse.cdt.lsp.internal.clangd.ClangdMetadataDefaults">
+   <service>
+      <provide interface="org.eclipse.cdt.lsp.clangd.ClangdMetadata"/>
+   </service>
+   <reference cardinality="1..1" field="defaults" interface="org.eclipse.cdt.lsp.clangd.ClangdOptions" name="defaults" target="(isDefaultOptions=true)"/>
```

Comment:
The main argument to have dedicated interface for defaults is to avoid this trick, and it can simplify custom component definition as well.

Reply:
Yes, I agree. Let's create an interface ClangdOptionsDefaults which extends ClangdOptions (which doesn't have `toList` anymore) but has no further methods. Then we can remove this trick.